File: gto-support-room/src/main/java/org/ccci/gto/android/common/room/converter/UriConverter.java
package org.ccci.gto.android.common.room.converter;

import android.net.Uri;

import androidx.annotation.Nullable;
import androidx.room.TypeConverter;

public class UriConverter {
    @Nullable
    @TypeConverter
    public static Uri toUri(@Nullable final String uri) {
        return uri != null ? Uri.parse(uri) : null;
    }

    @Nullable
    @TypeConverter
    public static String toString(@Nullable final Uri uri) {
        return uri != null ? uri.toString() : null;
    }
}

File: gto-support-room/src/main/java/org/ccci/gto/android/common/room/converter/UriConverter.kt
package org.ccci.gto.android.common.room.converter

import android.net.Uri
import androidx.room.TypeConverter

@Deprecated("Since v3.4.0, use converter from the gto-support-androidx-room module instead")
object UriConverter {
    @JvmStatic
    @TypeConverter
    fun toUri(uri: String?) = uri?.let { Uri.parse(uri) }

    @JvmStatic
    @TypeConverter
    fun toString(uri: Uri?) = uri?.toString()
}
